server/routes/bulkThreads.ts

```diff
@@ -3,11 +3,31 @@ import lookupCommunityIsVisibleToUser from '../util/lookupCommunityIsVisibleToUs
 import { UserRequest } from '../types';
 
 const bulkThreads = async (models, req: UserRequest, res: Response, next: NextFunction) => {
+  const { Op } = models.sequelize;
   const [chain, community] = await lookupCommunityIsVisibleToUser(models, req.query, req.user, next);
 
-  const threads = await models.OffchainThread.findAll({
-    where: community ? { community: community.id }
-      : chain ? { chain: chain.id } : {},
+  const userAddresses = await req.user.getAddresses();
+  const userAddressIds = Array.from(userAddresses.map((address) => address.id));
+  const roles = await models.Role.findAll({
+    where: community ? {
+      address_id: { [Op.in]: userAddressIds },
+      offchain_community_id: community.id,
+    } : chain ? {
+      address_id: { [Op.in]: userAddressIds },
+      chain_id: chain.id,
+    } : {},
+  });
+
+  const adminRoles = roles.filter((r) => r.permission === 'admin' || r.permission === 'moderator');
+
+  const allThreads = await models.OffchainThread.findAll({
```

Comment:
can we simplify this with if/else prior to passing it into the query?

Reply:
👍 factored out!